### PR TITLE
feat: [CI-9350]: use step params from ci manager

### PR DESCRIPTION
**Pull request by harshjain12**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52469)

### CODE REVIEW

**Comment by harshjain12** [comment]
trigger uts

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/execution/PipelineExecutionUpdateEventHandler.java [hunk 1]
```diff
@@ -212,12 +228,27 @@ private void sendGitStatus(
         if (isAutoAbortThroughTrigger(event)) {
           log.info("Skipping updating Git status as execution was Auto aborted by trigger due to newer execution");
         } else {
+          String runTimeId = AmbianceUtils.obtainCurrentRuntimeId(ambiance);
```
Comment (raghavharness) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/execution/PipelineExecutionUpdateEventHandler.java:231]: Can we move this to a common utility ? I see we are replicating this code 3 times.

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/IntegrationStageStepPMS.java [hunk 2]
```diff
@@ -108,6 +112,15 @@ public ChildExecutableResponse obtainChild(
     log.info("Executing integration stage with params accountId {} projectId {} [{}]", ngAccess.getAccountIdentifier(),
         ngAccess.getProjectIdentifier(), stepParameters);
 
+    String stageRuntimeId = AmbianceUtils.getStageRuntimeIdAmbiance(ambiance);
```
Comment (raghavharness) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/IntegrationStageStepPMS.java:115]: Can we move this to a common utility ? I see we are replicating this code 3 times.

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/codebase/CodeBaseTaskStep.java [hunk 3]
```diff
@@ -107,6 +112,17 @@ public TaskRequest obtainTask(
       throw new CIStageExecutionException("{} type is not supported in codebase delegate task for scm api operation");
     }
 
+    String runTime = AmbianceUtils.obtainCurrentRuntimeId(ambiance);
+    String accountId = AmbianceUtils.getAccountId(ambiance);
+    String stageRuntimeId = AmbianceUtils.getStageRuntimeIdAmbiance(ambiance);
+
+    stepExecutionParametersRepository.save(StepExecutionParameters.builder()
```
Comment (raghavharness) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/codebase/CodeBaseTaskStep.java:119]: Don't we need to do it for executeSync function as well in this class ?

**Review by raghavharness — CHANGES_REQUESTED** [review]

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/codebase/CodeBaseTaskStep.java (hunk 3 above)
Comment (harshjain12) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/codebase/CodeBaseTaskStep.java:119]: Done

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/execution/PipelineExecutionUpdateEventHandler.java (hunk 1 above)
Comment (harshjain12) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/execution/PipelineExecutionUpdateEventHandler.java:231]: it has state stepExecutionParametersRepository and it is just finding document with one method call so moving it to util might not help much.
Let me know if you still think we should move it.

**Review by raghavharness — DISMISSED** [review]

**Comment by harshjain12** [comment]
trigger uts

**Comment by harshjain12** [comment]
trigger uts

**Pull request merged**